### PR TITLE
id:86063 add tests for brandgility embedded api

### PR DESCRIPTION
**Pull request by wenzelua**

### CODE REVIEW

File: src/lib.test.js [hunk 1]
```diff
@@ -1,7 +1,96 @@
...
+
+  const messageEvent = new MessageEvent('message', {
+    data: { identityKey, action },
+    source: differentIframe.contentWindow,
+  });
+
+  window.dispatchEvent(messageEvent);
+
+  expect(callback).not.toHaveBeenCalled();
+});
```
Comment (sunstorymvp) [inline src/lib.test.js:52]: this test case is not complete, please listen for `action`
expectation is correct

File: src/lib.test.js [hunk 2]
```diff
@@ -1,7 +1,96 @@
...
+    data: { identityKey, action },
+    source: iframe.contentWindow,
+  });
+
+  window.dispatchEvent(messageEvent);
+
+  expect(callback).not.toHaveBeenCalled();
 });
+
+test('callback should NOT be called after destroy', () => {
```
Comment (sunstorymvp) [inline src/lib.test.js:69]: make sure that listeners are de-attached on destroy in separate test case

File: src/lib.test.js [hunk 3]
```diff
@@ -1,7 +1,96 @@
...
+    data: { identityKey, action },
+    source: iframe.contentWindow,
+  });
+
+  window.dispatchEvent(messageEvent);
+
+  expect(callback).not.toHaveBeenCalled();
+});
+
+test('callback should be called when emit is called with appropriate action', () => (
```
Comment (sunstorymvp) [inline src/lib.test.js]: you can use async/await in tests

File: src/lib.test.js (hunk 2 above)
Comment (wenzelua) [reply src/lib.test.js:69]: `handlersByAction` as well as `handleMessage` (post message handler) are private

Comment (sunstorymvp) [reply src/lib.test.js:69]: but you can check listeners attached on `message` event on target window

Comment (wenzelua) [reply src/lib.test.js:69]: `getEventListeners` is not a standard JavaScript function. It's an extension provided by Chrome DevTools

File: src/lib.test.js [hunk 4]
```diff
@@ -1,7 +1,103 @@
...
+  brandgilityEmbeddedApi.on(action, callback);
+
+  const messageEvent = new MessageEvent('message', {
+    data: { identityKey, action, payload },
+    source: iframe.contentWindow,
+  });
+
+  window.dispatchEvent(messageEvent);
+
+  expect(callback).toHaveBeenCalledWith(payload);
```
Comment (sunstorymvp) [inline src/lib.test.js:28]: please move this expectation to first test case, remove this one

**Review by ghost — APPROVED** [review]

**Review by sunstorymvp — APPROVED** [review]

File: src/lib.test.js [hunk 5]
```diff
@@ -1,7 +1,96 @@
...
+  const payload = {
+    someField: 'someValue',
+  };
+
+  brandgilityEmbeddedApi.on(action, callback);
+
+  const messageEvent = new MessageEvent('message', {
+    data: { identityKey, action, payload },
+    source: iframe.contentWindow,
+  });
```
Comment (sunstorymvp) [inline src/lib.test.js:22]: please move variable definitions on top before function calls, see other places

**Pull request merged**